packages/fields/src/types/Relationship/views/Controller.js

```diff
@@ -28,26 +28,43 @@ export default class RelationshipController extends FieldController {
     return `${this.getFilterLabel({ label })}: "${value}"`;
   };
 
-  // TODO: FIXME: This should be `set`, not `connect`
-  buildRelateToOneInput = ({ id }) => ({ connect: { id } });
-  buildRelateToManyInput = data => ({ connect: data.map(({ id }) => ({ id })) });
+  deserialize = data => {
+    // this is probably not a great solution
+    // ideally, we should have a set option in the mutation
+    this.lastSavedState = this.config.many
+      ? (data[this.path] || []).map(x => x.id)
+      : data[this.path]
+      ? data[this.path].id
+      : null;
+    return data[this.path];
+  };
 
   serialize = data => {
     const { path } = this;
     const { many } = this.config;
 
-    if (!data[path]) {
-      return many ? { disconnectAll: true } : null;
-    }
+    let value = data[path];
 
     if (many) {
-      if (!Array.isArray(data[path]) || !data[path].length) {
-        return { disconnectAll: true };
+      let ids = [];
+      if (Array.isArray(value)) {
+        ids = value.map(x => x.id);
       }
-      return this.buildRelateToManyInput(data[path], path);
+      return {
+        connect: ids.map(id => ({ id })),
+        disconnect: this.lastSavedState
+          .filter(id => {
+            return !ids.includes(id);
+          })
+          .map(id => ({ id })),
+      };
```

Comment:
The other option here is:

```javascript
return {
  disconnectAll: true,
  connect: ids.map(id => ({ id })),
};
```

This is [the equivalent of a `set`](https://v5.keystonejs.com/keystone-alpha/fields/src/types/relationship/#overriding-a-to-many-relationship).

Reply:
Also removes the need for storing the `lastSavedState`.

Reply:
Just tried that and it changes the order of the items which would probably be really confusing for users

Reply:
🤔 Hmm, it should set them in the order that they are in the array passed to the mutation. @timleslie we spoke about this the other day, and I think we landed on "it's up to the database to say what the order should be" IIRC, but Mongo should respect the order.

Reply:
Just opened an issue for this: #1362